### PR TITLE
docs: enhance documentation

### LINKED ISSUE #185: Enhance documentation
Follow same approach than for the ingest-manager test suite, for the other suites.

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
This PR is simplifies the existing docs, creating a main README file per test suite, moving some of the sections to other docs, and removing others because they are outdated or even present in a more up-to-date doc.

~~Finally, we added the CI badges for the scheduled pipeline for each test suite~~~ Edited: See https://github.com/elastic/e2e-testing/pull/191#discussion_r459612392 

## Why is it important?
CCC: Clearness, Conciseness and Concreteness

We are also preparing the project for transferring it to the functional teams

## Checklist

~~- [ ] My code follows the style guidelines of this project~~
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
- [x] I have made corresponding changes to the documentation
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
~~- [ ] I have run the Unit tests for the CLI, and they are passing locally~~
~~- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally~~
~~- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)~~

## Related issues

- Closes #185 
- Merge after #194

### CODE REVIEW

File: e2e/_suites/metricbeat/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+[![Build Status](https://beats-ci.elastic.co/buildStatus/icon?job=e2e-tests%2Fe2e-testing-integrations-daily)](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-integrations-daily/)
```
Comment (mdelapenya) [inline e2e/_suites/metricbeat/README.md:1]: Wdyt about propagating the build result from the downstream job to the scheduled one?

File: e2e/README.md [hunk 2]
```diff
@@ -1,81 +1,73 @@
...
+>Gherkin uses a set of special keywords to give structure and meaning to executable specifications. Each keyword is translated to many spoken languages. Most lines in a Gherkin document start with one of the keywords.
 
-### Runtime dependencies
-We want to store the metrics in Elasticsearch, so at some point we must start up an Elasticsearch instance. Besides that, we want to query the Elasticsearch to perform assertions on the metrics, such as there are no errors, or the field `f.foo` takes the value `bar`. For that reason we need an Elasticsearch in a well-known location. Here it appears the usage of the [Observability Provisioner CLI tool](../cli/README.md), which is a CLI writen in Go which exposes an API to query the specific runtime resources needed to run the tests. In our case, Metricbeat, we need just an Elasticsearch, but a Kibana could be needed in the case of verifying the dashboards are correct.
+The key part here is **executable specifications**: we will be able to automate the verification of the spefications anf potentially get a coverage of these specs.
```
Comment (v1v) [inline e2e/README.md:33]:
```suggestion
The key part here is **executable specifications**: we will be able to automate the verification of the specifications and potentially get a coverage of these specs.
```

File: e2e/_suites/helm/README.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
+[![Build Status](https://beats-ci.elastic.co/buildStatus/icon?job=e2e-tests%2Fe2e-testing-helm-daily)](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-helm-daily/)
```
Comment (v1v) [inline e2e/_suites/helm/README.md:1]: As long as the daily job does not wait for the downstream job's status then let's use the downstream one:

```suggestion
[![Build Status](https://beats-ci.elastic.co/buildStatus/icon?job=e2e-tests%2Fe2e-testing-mbp%2Fmaster)](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-mbp/job/master/)
```

File: e2e/_suites/helm/README.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+## Motivation
+
+Our goal is for the Observability team to execute this automated e2e test suite while developing the Helm charts for APM Server, Filebeat and Metricbeat. The tests in this folder assert that the use cases (or scenarios) defined in the `features` directory are behaving as expected.
+
+## How do the tests work?
+
+At the topmost level, the test framework uses a BDD framework written in Go, where we set
+the expected behavior of use cases in a feature file using Gherkin, and implementing the steps in Go code.
+The provisining of services is accomplish using [Kind (Kubernetes in Docker)](https://kind.sigs.k8s.io/https://kind.sigs.k8s.io/) and [Helm](https://helm.sh/) packages.
```
Comment (v1v) [inline e2e/_suites/helm/README.md:13]:
```suggestion
The provisioning of services is accomplished using [Kind (Kubernetes in Docker)](https://kind.sigs.k8s.io/https://kind.sigs.k8s.io/) and [Helm](https://helm.sh/) packages.
```

File: e2e/_suites/helm/README.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+   - Install godog: `make -C e2e install-godog`
+
+4. Run the tests.
+
+   ```shell
+   cd e2e/_suites/helm
+   OP_LOG_LEVEL=DEBUG godog
+   ```
+
+   The tests will take a few minutes to run, spinning up the kubernetes cluster, instgalling the helm charts, and performing the test steps outlined earlier.
```
Comment (v1v) [inline e2e/_suites/helm/README.md:57]:
```suggestion
   The tests will take a few minutes to run, spinning up the Kubernetes cluster, installing the helm charts, and performing the test steps outlined earlier.
```

File: e2e/_suites/ingest-manager/README.md [hunk 6]
```diff
@@ -1,3 +1,5 @@
+[![Build Status](https://beats-ci.elastic.co/buildStatus/icon?job=e2e-tests%2Fe2e-testing-ingest-manager-daily)](https://beats-ci.elastic.co/job/e2e-tests/job/e2e-testing-ingest-manager-daily/)
```
Comment (v1v) [inline e2e/_suites/ingest-manager/README.md:1]: ditto

File: e2e/_suites/metricbeat/README.md (hunk 1 above)
Comment (v1v) [reply e2e/_suites/metricbeat/README.md:1]: Yep, I didn't see your comment and I commented the same in `e2e/_suites/helm/README.md` :)

Comment (mdelapenya) [reply e2e/_suites/metricbeat/README.md:1]: It's a matter of adding `propagate: true` to the trigger, right? What if the `wait` is set to false? Will Jenkins update the job status after downstream job finishes?

Comment (v1v) [reply e2e/_suites/metricbeat/README.md:1]: IIRC, propagate will force the wait, but no the other way around

I'd say using the downstream build badge gets some other benefits:
- one worker only for the running, aka, the upstream does not need to block the worker to wait for the downstream build.
- parenstream does not need to populate the test results from the downstream.
- easy to debug what's going in the logs

Comment (mdelapenya) [reply e2e/_suites/metricbeat/README.md:1]: But how to know if the proper build for the scheduled build is green? I can imagine a scheduled execution failing, say it for Helm, and after that the metricbeat suite triggers a green build. Then the Helm badge will point to the status of a green downstream build, which could be misleading

Comment (mdelapenya) [reply e2e/_suites/metricbeat/README.md:1]: @elastic/observablt-robots I'm going to remove the badges, merge, and open a discussion around badges and upstream/downstream jobs.

Thanks!

**Pull request merged**